pkg/client/query/sharedquerier.go

```diff
@@ -105,3 +108,66 @@ func (sq *sharedQuerier) GetSessionGracePeriodEndHeight(
 	}
 	return shared.GetSessionGracePeriodEndHeight(sharedParams, queryHeight), nil
 }
+
+// GetEarliestClaimCommitHeight returns the earliest block height at which a claim
+// for the session that includes queryHeight can be committed for a given supplier.
+//
+// TODO_TECHDEBT(#543): We don't really want to have to query the params for every method call.
+// Once `ModuleParamsClient` is implemented, use its replay observable's `#Last()` method
+// to get the most recently (asynchronously) observed (and cached) value.
+// TODO_BLOCKER(@bryanchriswhite, #543): We also don't really want to use the current value of the params.
+// Instead, we should be using the value that the params had for the session which includes queryHeight.
+func (sq *sharedQuerier) GetEarliestClaimCommitHeight(ctx context.Context, queryHeight int64, supplierAddr string) (int64, error) {
```

Comment:
OPTIONAL NIT: multiple method header

Reply:
I'm not sure what the ask is here. Can you elaborate?